### ISSUE TITLE
Support multi-cluster Node DNS resolution

### ISSUE THREAD

**Issue by andrewsykim** [vmware-tanzu/cross-cluster-connectivity#71]
## Description
Support DNS resolution of Nodes across clusters.

## Desired Solution
It would be useful if nodes can be resolved across clusters through one of the following domains:

1) to discover an arbitrary selection of nodes in another cluster
```
nodes.<cluster-name>.<cluster-namespace>.svc.clusters.local
``` 

2) to discover a specific node with a well-known name:
```
<my-special-node>.nodes.<cluster-name>.<cluster-namespace>.svc.clusters.local
```

1) would be useful if an application can only be accessed via node ports (e.g. L4 only) and it doesn't make sense to go through an ingress gateway. 2) is useful to connect to workloads that may be bound to specific nodes in a cluster (e.g. a node that has a mounted volume or device and a stateful pod is always scheduled there). 

## Alternatives Considered
N/A

## Additional Context
N/A

**Comment by rosenhouse** [comment]
Yes, and then `nodes.<cluster-name>...:<node-port>` could be a backend `Endpoint` for a selectorless service.  Which means we could use this as a data plane for [MCS](https://github.com/kubernetes-sigs/mcs-api).  😄

**Comment by rosenhouse** [comment]
We're going to do #99 as a first cut here. We can do `<my-special-node>` as a follow-on.

A different follow-on might be label selector, based?  We should look at use cases before deciding
